lib/streamlit/elements/image.py

```diff
@@ -386,6 +387,42 @@ def image_to_url(
         return ""
 
 
+def _get_svg_width_and_height(image: str) -> Tuple[str, str]:
+    root_idx = image.find(">")
+    root = image[:root_idx]
+    width = "100"
+    height = "100"
+    vb = root.find("viewBox")
+    if vb != -1:
+        view_box = [
+            val.replace('"', "") for val in root[vb + len("viewBox=") + 1 :].split(" ")
+        ][:4]
+        width, height = view_box[2], view_box[3]
```

Comment:
Unfortunately, I don't have a lot of knowledge about SVGs. But are we sure that we can always just extract the `width` & `height` from the viewbox and apply inject as `width` & `height` into the SVG?

Reply:
We already have some logic for setting the dimensions of normal images on the frontend, which also works for `use_column_width` and fullscreen mode here:

https://github.com/streamlit/streamlit/blob/5edf0aba3a828082457427e7c5f6d7f5f0943f2f/frontend/src/components/elements/ImageList/ImageList.tsx#L59

I assume it would make things more consistent if we also reuse these style settings for SVGs. However, looking at the PR, it seems to anyways already use the frontend based styling when the user sets the `width` (since it is sent via the `url` parameter and thereby rendered via the `img` tag). But I'm wondering if we actually even need to set `width` and `height` in the SVG in this case?

Reply:
Frontend now has really nice CSS setup for width autoscaling for `img` HTML tag, which works very well. I'd like to enforce on backend side to render SVGs as `img` whenever that's possible.

Reply:
So, why not fully remove the support for `markup` and always render SVGs in an `img` tag? Otherwise, I'm a bit worried that we have different ways on how we deal with SVGs. However, there are some differences on how SVGs are handled when used directly in HTML or when used via `img` tag (see [here](https://stackoverflow.com/questions/66720281/svg-code-within-html-versus-svg-embedded-with-img-tag)).

Reply:
> However, there are some differences on how SVGs are handled when used directly in HTML or when used via img tag

Yes, that's the problem, especially quoting from [here](https://stackoverflow.com/questions/66720281/svg-code-within-html-versus-svg-embedded-with-img-tag):

> I believe when an svg uses external files like fonts or bitmaps it doesn't work when used as an <img> for security reasons. On the other hand it's easier to resize when used as <img> using css, from what I remember.

We already have a test case for SVGs containing external content and I didn't manage to get it work with `img` tag, I'd love to completely rid of `markup`. 

For now I don't see a way to make all SVGs compatible with our scaling of `img`, but technically that should be possible.

This PR makes these SVGs which can be rendered as `img` -> rendered as `img`